src/main.rs

```diff
@@ -45,11 +40,11 @@ impl cli::IntoExit for Worker {
 		let (exit_send, exit) = oneshot::channel();
 
 		let exit_send_cell = RefCell::new(Some(exit_send));
-		ctrlc::CtrlC::set_handler(move || {
+		ctrlc::set_handler(move || {
 			if let Some(exit_send) = exit_send_cell.try_borrow_mut().expect("signal handler not reentrant; qed").take() {
 				exit_send.send(()).expect("Error sending exit notification");
 			}
-		});
+		}).expect("Error setting Ctrl-C handler");
```

Comment:
any way we can exit gracefully in this case?

Reply:
You mean, if we didn't `expect` here but continued?

I don't think so considering that the ctrl-c doc that it errors 
> if another `ctrlc::set_handler()` handler exists or if a system error occurred while setting the handler.

The previous is clearly a programmer error the later is ... well, serious...